### PR TITLE
xds: Add Kafka rules into NetworkPolicy

### PR DESCRIPTION
**Pull request by rlenglet**
Signed-off-by: Romain Lenglet <romain@covalent.io>

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/4567)

### CODE REVIEW

**Review by aanm — APPROVED** [review]

**Comment by aanm** [comment]
test-me-please

**Review by ianvernon — APPROVED** [review]

**Review by jrajahalme — APPROVED** [review]
LGTM

File: envoy/cilium/npds.proto [hunk 1]
```diff
@@ -115,3 +120,36 @@ message HttpNetworkPolicyRule {
...
+  // The set of Kafka network policy rules.
+  // A Kafka request is matched if any of its rules matches the request.
+  // Required and may not be empty.
+  repeated KafkaNetworkPolicyRule kafka_rules = 1 [(validate.rules).repeated.min_items = 1];
+}
+
+// A Kafka network policy rule, as a conjunction of predicates on Kafka requests.
+// If all the predicates of a rule match a Kafka request, the request is allowed. Otherwise, it is
+// denied.
+message KafkaNetworkPolicyRule {
```
Comment (manalibhutiyani) [inline envoy/cilium/npds.proto:135]: Should we also include Kafka `Roles` or that will already be resolved to low level apiKeys when we check these values?

Comment (rlenglet) [reply envoy/cilium/npds.proto:135]: The roles will be resolved when converting into those rules.

**Pull request merged**